offixapp/src/App.js

```diff
@@ -18,40 +18,52 @@ const App = () => {
     return (
       <ApolloOfflineProvider client={offlineClient}>
         <ApolloProvider client={offlineClient}>
-          <ShopScreen></ShopScreen>
+          <ShopScreen />
         </ApolloProvider>
       </ApolloOfflineProvider>
     )
   }
   return <Text>Loading</Text>
 };
 
-const ShopComponent = withQuery(shopsQuery, { options: { fetchPolicy: "cache-and-network" } })(props => {
-  const { error, findAllShops } = props.data;
-  if (error) {
-    console.log(error);
-    return <Text>Error when querying data</Text>;
-  }
-  if (findAllShops) {
-    return (
-      <View>
-        {findAllShops.map((shop, index) => {
-          return <Text key={index}>{shop.name}</Text>;
-        })}
-      </View>
-    );
-  }
+const ShopComponent = forwardRef((props, ref) => {
+
+  const { error, data, loading } = useQuery(shopsQuery, { options: { fetchPolicy: "cache-and-network" }})
 
-  return <Text>Loading...</Text>;
+  console.log(JSON.stringify(loading ? "Loading..." : "Loaded."))
+
+  if ( loading === true ) {
+    return <Text>Loading...</Text>;
+  } else {
+
+    if (error) {
+      console.log(error);
+      return <Text>Error when querying data</Text>;
+    } else {
+      const { findAllShops } = data;
+      if (findAllShops) {
+        return (
+          <View>
+            {findAllShops.map((ref, index) => {
+              return <Text key={index}>{ref.name}</Text>;
+            })}
+          </View>
+        );
+      }
+    }
+  }
 });
 
 
 export function ShopScreen() {
-  const [shop, setShop] = useState({
+
+  const shopRef = useRef({
```

Comment:
@mmusil if you try using `createRef` instead of `useRef`, apparently `createRef` resents the reference to its initial value after render. This might fix the issue you mentioned